### ISSUE TITLE
about:sync is very hard to read, if you're using Firefox 96 or higher in dark mode

### ISSUE THREAD

**Issue by unforgettableid** [mozilla-extensions/aboutsync#78]
TL;DR:  Please force `about:sync` to always use a white background.

## The problem

In Firefox 96 and higher, when your OS is set to dark mode, Firefox uses a dark gray background on various UI pages, including about:sync.  (Based on [this source](https://support.mozilla.org/en-US/questions/1364252).)

Unfortunately, about:sync is very hard to read when it has a dark gray background.

For example:

*  Black text is very difficult to read.
*  White text on a light gray background (e.g. "Please describe how you intend using about:sync") is very difficult to read.
*  Blue text (e.g. tab lastUsed timestamps) are difficult to read.

## Recommended contrast ratios

For most text, the WCAG recommends a minimum contrast ratio of at least 4.5:1 or better.  Ideally, they recommend a contrast ratio of 7:1 or better.  [(Source.)](https://developer.mozilla.org/en-US/docs/Web/Accessibility/Understanding_WCAG/Perceivable/Color_contrast)

To check the contrast ratios of any webpage, right-click on it and cloose "Inspect".  Now click "Accessibility" or press Shift+F12.  Finally, next to "Check for Issues", change "None" to "Contrast".

## A contrast fixer tool

Text with insufficient contrast can be difficult to read for everyone, but especially for people with low vision.  If you do a Web search for contrast fixer tools, you'll find lots of web apps; [Contrast Finder](https://app.contrast-finder.org/) is a good one.  It will modify any chosen pair of colors, in order to increase the contrast ratio.

## Steps to reproduce

One way to reproduce the issue is to simply set your OS to dark mode.

The other way to reproduce the issue is to tell Firefox to pretend that your OS is set to dark mode.  Here's how:

*  Visit `about:config`.
*  If `layout.css.prefers-color-scheme.content-override` is set, note down its value for future reference.
*  Set `layout.css.prefers-color-scheme.content-override` to 0.
*  Visit `about:sync`.
*  Click the Reload button.  The issue should immediately become apparent.

Once you're finished looking at the problem, go back to `about:config`, and click the Reset button to restore `layout.css.prefers-color-scheme.content-override` to its default value (probably 3).

## Recommended solution

The simplest solution is probably to modify `about:sync` to ignore dark mode, and to always use a white background everywhere.  The current `about:sync` color palette provides mostly-acceptable contrast ratios over top of a white background.  If you decide to allow dark backgrounds, you might need to revamp the palette.
